### PR TITLE
Bugfix to testScreenshots issue #13 

### PR DESCRIPTION
**Pull request by dexfs**
I made some adjusts on project and fixed testScreenshots. 

As you see on http://prntscr.com/ex5lk1

### CODE REVIEW

**Comment by rkcudjoe** [comment]
@dexfs are you available to fix the rest of the issues mentioned in #13?

**Comment by dexfs** [comment]
@rkcudjoe I a busy a little bit but I can try work on 
1 - tests should be name NameTest e.g. ActivitiesTest and the file should be ActivitiesTest.php

The 4  I think I'm not understood the problem and I don't have experience with mocks, but I can try and if I got something I send a PR again

File: .gitignore [hunk 1]
```diff
@@ -1 +1,3 @@
+.idea
+/vendor
```
Comment (malukenho) [inline .gitignore]: `vendor/`

File: .gitignore [hunk 2]
```diff
@@ -1 +1,3 @@
+.idea
```
Comment (malukenho) [inline .gitignore]: Just put it on your global config

File: composer.json [hunk 3]
```diff
@@ -1,18 +1,20 @@
 {
-
-    "require": {
-        "ext-curl": "*",
-        "beberlei/assert": "2.*",
-        "symfony/yaml": "2.*|3.*",
-        "symfony/event-dispatcher": "~2.1|3.*"
+  "require": {
```
Comment (malukenho) [inline composer.json]: doesn't need to re-format all the file as it's not part of the bugfix.

File: composer.json [hunk 4]
```diff
@@ -1,18 +1,20 @@
...
+    "ext-curl": "*",
```
Comment (malukenho) [inline composer.json]: no minimum php version?

File: config.php [hunk 5]
```diff
@@ -1,48 +1,46 @@
...
+require_once(ROOT_FOLDER . "lib/auth.php");
+require_once(ROOT_FOLDER . "lib/users.php");
+require_once(ROOT_FOLDER . "lib/organizations.php");
+require_once(ROOT_FOLDER . "lib/projects.php");
+require_once(ROOT_FOLDER . "lib/activities.php");
+require_once(ROOT_FOLDER . "lib/screenshots.php");
+require_once(ROOT_FOLDER . "lib/notes.php");
+require_once(ROOT_FOLDER . "lib/weekly.php");
+require_once(ROOT_FOLDER . "lib/custom.php");
+require_once(ROOT_FOLDER . "helper/curl.php");
```
Comment (malukenho) [inline config.php]: EOL EOF

File: tests/screenshots.php [hunk 6]
```diff
@@ -1,25 +1,32 @@
...
\ No newline at end of file
+        $this->stub = $this->getMockBuilder('Hubstaff\Client')
+            ->disableOriginalConstructor()
+            ->getMock();
+
+    }
+
+    public function testScreenshots()
+    {
+        $starttime = "2016-03-14";
```
Comment (malukenho) [inline tests/screenshots.php]: single quotes

File: tests/screenshots.php [hunk 7]
```diff
@@ -1,25 +1,32 @@
...
+        $stoptime = "2016-03-20";
```
Comment (malukenho) [inline tests/screenshots.php]: single quotes

File: tests/screenshots.php [hunk 8]
```diff
@@ -1,25 +1,32 @@
...
+
+        $this->stub->expects($this->any())
```
Comment (malukenho) [inline tests/screenshots.php]: `self::any()`

Comment (malukenho) [inline tests/screenshots.php]: Just mock the object within

File: tests/screenshots.php [hunk 9]
```diff
@@ -1,25 +1,32 @@
...
+            ->method('screenshots')
+            ->will($this->returnValue(json_decode('{"screenshots":[]}', true)));
+
+        $this->assertArrayHasKey("screenshots", $this->stub->screenshots($starttime, $stoptime, $this->options, 0));
```
Comment (malukenho) [inline tests/screenshots.php]: `self::assertArrayHasKey` — make sure y're calling static method as static

File: tests/screenshots.php [hunk 10]
```diff
@@ -1,25 +1,32 @@
...
+class test_screenshots extends \PHPUnit_Framework_TestCase
+{
+    public $stub;
+    public $options = [];
 
-	    $this->stub->method('screenshots')->willReturn(json_decode('{"screenshots":[]}',true));	
-	    $this->assertArrayHasKey("screenshots", $this->stub->screenshots($starttime, $stoptime, $this->options, 0));
-		}
-	}
+    public function __construct()
```
Comment (malukenho) [inline tests/screenshots.php]: No no. use `setUp` method

File: tests/screenshots.php [hunk 11]
```diff
@@ -1,25 +1,32 @@
...
+    {
+        parent::__construct();
```
Comment (malukenho) [inline tests/screenshots.php]: No

File: tests/screenshots.php [hunk 12]
```diff
@@ -1,25 +1,32 @@
...
+        $this->options["users"] = "61188";
+        $this->options["projects"] = "112761";
+        $this->options["organizations"] = "27572";
 
-?>
\ No newline at end of file
+        $this->stub = $this->getMockBuilder('Hubstaff\Client')
```
Comment (malukenho) [inline tests/screenshots.php]: `Hubstaff\Client::class`

File: tests/screenshots.php [hunk 13]
```diff
@@ -1,25 +1,32 @@
...
+            ->disableOriginalConstructor()
+            ->getMock();
+
```
Comment (malukenho) [inline tests/screenshots.php]: remove empty line

File: tests/screenshots.php [hunk 14]
```diff
@@ -1,25 +1,32 @@
...
-			require_once("hubstaff.php");
-	    $this->stub = $this->getMockBuilder('hubstaff\Client')->disableOriginalConstructor()->getMock();
-		}		
-		public function testScreenshots()
-		{
-      $starttime = "2016-03-14";
-      $stoptime = "2016-03-20";
+class test_screenshots extends \PHPUnit_Framework_TestCase
+{
+    public $stub;
```
Comment (malukenho) [inline tests/screenshots.php]: no docs?

File: tests/screenshots.php [hunk 15]
```diff
@@ -1,25 +1,32 @@
...
+    public $options = [];
```
Comment (malukenho) [inline tests/screenshots.php]: docs?

File: tests/screenshots.php [hunk 16]
```diff
@@ -1,25 +1,32 @@
...
+<?php
 
-			require_once("hubstaff.php");
-	    $this->stub = $this->getMockBuilder('hubstaff\Client')->disableOriginalConstructor()->getMock();
-		}		
-		public function testScreenshots()
-		{
-      $starttime = "2016-03-14";
-      $stoptime = "2016-03-20";
+class test_screenshots extends \PHPUnit_Framework_TestCase
```
Comment (malukenho) [inline tests/screenshots.php]: `final` and import `PHPUnit_Framework_TestCase` namespace

File: tests/bootstrap.php [hunk 17]
```diff
@@ -1,3 +1,2 @@
 <?php
-require_once __DIR__ . 'vendor/autoload.php';
-?>
\ No newline at end of file
+require_once __DIR__ . '/../vendor/autoload.php';
```
Comment (malukenho) [inline tests/bootstrap.php]: may be you can delete it and point it out on `phpunit.xml`

File: hubstaff.php [hunk 18]
```diff
@@ -1,163 +1,172 @@
...
-		}
-	}
-
-?>
\ No newline at end of file
+<?php
+
+namespace Hubstaff;
+
+class Client
```
Comment (malukenho) [inline hubstaff.php]: no interface?

File: hubstaff.php [hunk 19]
```diff
@@ -1,163 +1,172 @@
...
+{
+    public $app_token = "";
```
Comment (malukenho) [inline hubstaff.php]: docs?

File: hubstaff.php [hunk 20]
```diff
@@ -1,163 +1,172 @@
...
+    public $auth_token = "";
```
Comment (malukenho) [inline hubstaff.php]: docs?

File: hubstaff.php [hunk 21]
```diff
@@ -1,163 +1,172 @@
...
+    }
+
+    public function set_auth_token($auth_token)
+    {
+        $this->auth_token = $auth_token;
+    }
+
+    public function auth($email, $password)
+    {
+        $auth = new Client\userauth;
```
Comment (malukenho) [inline hubstaff.php]: Hidden dependency, pass it through the constructor or via method parameter.

File: hubstaff.php [hunk 22]
```diff
@@ -1,163 +1,172 @@
...
+        $auth_token = $auth->auth($this->app_token, $email, $password, BASE_URL . AUTH);
+        if (isset($auth_token["error"])) {
+            return $auth_token["error"];
+        }
+        $this->set_auth_token($auth_token["auth_token"]);
+    }
+
+    public function users($organization_memberships = 0, $project_memberships = 0, $offset = 0)
+    {
+        $users = new Client\users;
```
Comment (malukenho) [inline hubstaff.php]: Hidden dependency, pass it through the constructor or via method parameter.

File: hubstaff.php [hunk 23]
```diff
@@ -1,163 +1,172 @@
...
+        return $users->getusers($this->auth_token, $this->app_token, $organization_memberships, $project_memberships, $offset, BASE_URL . USERS);
+    }
+
+    public function find_user($id)
+    {
+        $users = new Client\users;
```
Comment (malukenho) [inline hubstaff.php]: Hidden dependency, pass it through the constructor or via method parameter.

File: hubstaff.php [hunk 24]
```diff
@@ -1,163 +1,172 @@
...
+        return $users->find_user($this->auth_token, $this->app_token, sprintf(BASE_URL . FIND_USER, $id));
+    }
+
+    public function find_user_orgs($id, $offset = 0)
+    {
+        $users = new Client\users;
```
Comment (malukenho) [inline hubstaff.php]: Hidden dependency, pass it through the constructor or via method parameter.

File: hubstaff.php [hunk 25]
```diff
@@ -1,163 +1,172 @@
...
+        return $users->find_user_orgs($this->auth_token, $this->app_token, $offset, sprintf(BASE_URL . FIND_USER_ORG, $id));
+    }
+
+    public function find_user_projects($id, $offset = 0)
+    {
+        $users = new Client\users;
```
Comment (malukenho) [inline hubstaff.php]: Hidden dependency, pass it through the constructor or via method parameter.

File: hubstaff.php [hunk 26]
```diff
@@ -1,163 +1,172 @@
...
+        return $users->find_user_projects($this->auth_token, $this->app_token, $offset, sprintf(BASE_URL . FIND_USER_PROJ, $id));
+    }
+
+    public function organizations($offset = 0)
+    {
+        $organizations = new Client\organizations;
```
Comment (malukenho) [inline hubstaff.php]: Hidden dependency, pass it through the constructor or via method parameter.

File: hubstaff.php [hunk 27]
```diff
@@ -1,163 +1,172 @@
...
+        return $organizations->getorganizations($this->auth_token, $this->app_token, $offset, BASE_URL . ORGS);
+    }
+
+    public function find_organization($id)
+    {
+        $organizations = new Client\organizations;
```
Comment (malukenho) [inline hubstaff.php]: Hidden dependency, pass it through the constructor or via method parameter.

File: hubstaff.php [hunk 28]
```diff
@@ -1,163 +1,172 @@
...
+        return $organizations->find_organization($this->auth_token, $this->app_token, sprintf(BASE_URL . FIND_ORG, $id));
+    }
+
+    public function find_org_projects($id, $offset = 0)
+    {
+        $organizations = new Client\organizations;
```
Comment (malukenho) [inline hubstaff.php]: Hidden dependency, pass it through the constructor or via method parameter.

File: hubstaff.php [hunk 29]
```diff
@@ -1,163 +1,172 @@
...
+        return $organizations->find_org_projects($this->auth_token, $this->app_token, $offset, sprintf(BASE_URL . FIND_ORG_PROJ, $id));
+    }
+
+    public function find_org_members($id, $offset = 0)
+    {
+        $organizations = new Client\organizations;
```
Comment (malukenho) [inline hubstaff.php]: Hidden dependency, pass it through the constructor or via method parameter.

File: hubstaff.php [hunk 30]
```diff
@@ -1,163 +1,172 @@
...
+        return $organizations->find_org_members($this->auth_token, $this->app_token, $offset, sprintf(BASE_URL . FIND_ORG_MEMBERS, $id));
+    }
+
+    public function projects($active = '', $offset = 0)
+    {
+        $projects = new Client\projects;
```
Comment (malukenho) [inline hubstaff.php]: Hidden dependency, pass it through the constructor or via method parameter.

File: hubstaff.php [hunk 31]
```diff
@@ -1,163 +1,172 @@
...
+        return $projects->getprojects($this->auth_token, $this->app_token, $active, $offset, BASE_URL . PROJS);
+    }
+
+    public function find_project($id)
+    {
+        $projects = new Client\projects;
```
Comment (malukenho) [inline hubstaff.php]: Hidden dependency, pass it through the constructor or via method parameter.

File: hubstaff.php [hunk 32]
```diff
@@ -1,163 +1,172 @@
...
+        return $projects->find_project($this->auth_token, $this->app_token, sprintf(BASE_URL . FIND_PROJ, $id));
+    }
+
+    public function find_project_members($id, $offset = 0)
+    {
+        $projects = new Client\projects;
```
Comment (malukenho) [inline hubstaff.php]: Hidden dependency, pass it through the constructor or via method parameter.

File: hubstaff.php [hunk 33]
```diff
@@ -1,163 +1,172 @@
...
+        $projects->find_project_members($this->auth_token, $this->app_token, $offset, sprintf(BASE_URL . FIND_PROJ_MEMBERS, $id));
+    }
+
+    public function activities($start_time, $stop_time, $offset = 0, $options = [])
+    {
+        $activities = new Client\activities;
```
Comment (malukenho) [inline hubstaff.php]: Hidden dependency, pass it through the constructor or via method parameter.

File: config.php [hunk 34]
```diff
@@ -1,48 +1,46 @@
...
-	require_once(ROOT_FOLDER."lib/notes.php");
-	require_once(ROOT_FOLDER."lib/weekly.php");
-	require_once(ROOT_FOLDER."lib/custom.php");
-	require_once(ROOT_FOLDER."helper/curl.php");
-		
-?>
+<?php
+
+define("ROOT_FOLDER", __DIR__ . "/");
+define("BASE_URL", "https://api.hubstaff.com/v1/");
```
Comment (malukenho) [inline config.php]: May be it should be moved to a config object or building

File: tests/screenshots.php (hunk 8 above)
Comment (dexfs) [inline tests/screenshots.php]: I understood, can you explain more about it ?

Comment (malukenho) [inline tests/screenshots.php]: @dexfs That's necessary once you're using `__construct` as a reliable thing (TIP: It's not, and you should get rid of it). `__construct` is only call one time when creating an object, the `setUp` method is called each time before a test is ran. Now, if you configure `$this->stub` on a first test method and on the second one you configure it as well but expecting something different, ya got a problem, in other words, your tests is not idem-potency.

**Pull request closed without merging**